sunpy/map/maputils.py

```diff
@@ -222,11 +245,14 @@ def coordinate_is_on_solar_disk(coordinates):
         Returns `True` if the coordinate is on disk, `False` otherwise.
     """
 
-    if not isinstance(coordinates.frame, Helioprojective):
+    # Calculate the coordinate's angular distance
+    coordinate_angles = np.sqrt(coordinates.Tx ** 2 + coordinates.Ty ** 2)
+    if _is_coordinate_helioprojective(coordinates):
+        # Calculate the angle of every pixel from the center of the Sun and compare it the angular
+        # radius of the Sun.
+        return coordinate_angles < solar_angular_radius(coordinates)
+    else:
         raise ValueError('The input coordinate(s) must be in the Helioprojective Cartesian frame.')
```

Comment:
```suggestion
    _verify_coordinate_helioprojective(coordinates)
    # Calculate the angle of every pixel from the center of the Sun and compare it the angular
    # radius of the Sun.
    return np.sqrt(coordinates.Tx ** 2 + coordinates.Ty ** 2) < solar_angular_radius(coordinates)
```

Reply:
Incidentally, your code would not work as intended because you moved the Helioprojective check to be *after* the access the attributes `.Tx` and `.Ty`, so that line would raise the error first.